nodejs/src/index.js

```diff
@@ -4,39 +4,55 @@ var TCompactProtocol = require('thrift/lib/nodejs/lib/thrift/compact_protocol');
 
 var Compression = exports.Compression = {
 	None: 0,
-	Gzip: 1
+	Gzip: 1,
+  Disable: 2
 };
 
 exports.write = function (struct, compression, callback) {
-	switch (compression || Compression.None) {
-		case Compression.None:
-			serialize(struct, callback);
-			break;
-		case Compression.Gzip:
-			zip(struct, callback);
-			break;
-		default:
-			callback(new Error('Unable to understand the compression applied'));
-			break;
-	}
+
+    switch (compression || Compression.None) {
+        case Compression.Disable:
+            serialize(struct, callback, true);
+        case Compression.None:
+            serialize(struct, callback);
+            break;
+        case Compression.Gzip:
+            zip(struct, callback);
+            break;
+        default:
+            callback(new Error('Unable to understand the compression applied'));
+            break;
+    }
 };
 
-exports.read = function (Model, rawData, callback) {
-	var buffer = Buffer.isBuffer(rawData) ? rawData : new Buffer(rawData, 'base64');
-
-	switch (buffer.readInt8(0)) {
-		case Compression.None:
-			readBytes(buffer.slice(1), Model, callback);
-			break;
-		case Compression.Gzip:
-			unzip(buffer.slice(1), Model, callback);
-			break;
-		default:
-			callback(new Error('Unable to understand the compression applied'));
-			break;
-	}
+exports.read = function (Model, rawData) {
```

Comment:
can you leave `callback` in the signature please? It's easier to read for people looking at the code.

Then please split arguments parsing from the actual logic. Maybe something like

```js
var callback = arguments.length === 4 ? arguments[4] : arguments[3];
var parseCompression = arguments.length == 4 ? () => Compression.Disabled : (buffer) => buffer.readInt8(0);

switch(parseCompression(buffer)) {
  case ...
}
```